### ISSUE TITLE
`health` endpoint states *syncing* even if synced

### ISSUE THREAD

**Issue by kkokosa** [NethermindEth/nethermind#6512]
**Description**
The `health` endpoint answers with `"IsSyncing": true` and `"description": "The node is still syncing, CurrentBlock: 18989326, HighestBlock: 18989327. Peers: 100.",` but the node is synced - for many hours the logs contain only repeating messages:

```
12 Jan 08:45:24 | Received new block:  18989411 (0x230a30...662a9d)
12 Jan 08:45:25 | Processed            18989411     |    537.37 ms  |  slot     11,729 ms | Gas gwei: 22.15 .. 22.36 (23.79) .. 57.66
12 Jan 08:45:25 | - Block              10.91 MGas   |    177    txs |  calls    517 ( 18) | sload   1,816 | sstore    541 | create   0
12 Jan 08:45:25 | - Block throughput   20.30 MGas/s |    329.39 t/s |          1.86 Blk/s | recv        0 | proc        0
12 Jan 08:45:25 | Received ForkChoice: Head: 18989411 (0x230a30...662a9d), Safe: 18989374 (0x4ea83d...cc330b), Finalized: 18989343 (0x30dcd8...d6fc59)
12 Jan 08:45:25 | Synced chain Head to 18989411 (0x230a30...662a9d)
```

**Steps to Reproduce**
Just run the non-validator mainnet or gearly fully-synced node and hit `health` JSON RPC endpoint.

**Screenshots**
Debugging shows it is "downloading receipts":
![image](https://github.com/NethermindEth/nethermind/assets/11159779/4d437170-b7ac-4b51-a9ea-248bcdfb9d08)
while it is not visible in logs etc.

**Configuration:**
- Windows, non-validator node, v1.25.0, Lighthouse  v4.5.0-441fc16

**Comment by marcindsobczak** [comment]
Was it synced on older version and then updated to 1.25.0?

**Comment by kkokosa** [comment]
Yes, from fully synced 1.21.0 that was then not running for a month or so.

**Comment by marcindsobczak** [comment]
up to 1.24.0 we used to have ancient barriers in syncing bodies and receipts. In 1.25.0 we removed ancient barriers, freshly synced nodes should download all bodies/receipts, but there is also a blockade to not download additional bodies/receipts on already synced and only updated to 1.25.0 nodes. This way in the line https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Facade/Eth/EthSyncingInfo.cs#L66
_blockTree.LowestInsertedBodyNumber = 11052984 (ancient barrier)
_syncConfig.AncientBodiesBarrierCalc = 1 (new barrier from config)

This way it is false and it will never change, as any bodies/recipts will not be downloaded. We will include fix for it in hotfix to 1.25.0

For now there is workaround for it to not receive `syncing` - you need to add ancient barriers back to `mainnet.cfg`

```
"Sync": {
  "AncientBodiesBarrier": 11052984,
  "AncientReceiptsBarrier": 11052984
}
```

or as running arguments `--Sync.AncientBodiesBarrier=11052984 --Sync.AncientReceiptsBarrier=11052984`
